src/modules/control_allocator/ActuatorEffectiveness/ActuatorEffectivenessHelicopterTest.cpp

```diff
@@ -0,0 +1,93 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <gtest/gtest.h>
+#include "ActuatorEffectivenessHelicopter.hpp"
+
+using namespace matrix;
+
+TEST(FunctionsTest, ThrottleCurve)
+{
+	ActuatorEffectivenessHelicopter helicopter(nullptr);
+	// run getEffectivenessMatrix with empty configuration to correctly initialize _first_swash_plate_servo_index
+	ActuatorEffectiveness::Configuration configuration{};
+	EffectivenessUpdateReason external_update = EffectivenessUpdateReason::MOTOR_ACTIVATION_UPDATE;
+	helicopter.getEffectivenessMatrix(configuration, external_update);
+
+	Vector<float, 6> control_sp{};
+	ActuatorEffectiveness::ActuatorVector actuator_sp{};
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = 0.1f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), 0.f);
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = 0.f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), 0.f);
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = -.125f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), .15f);
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = -.25f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), .3f);
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = -.375f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), .45f);
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = -.5f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), .6f);
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = -.625f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), .7f);
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = -.75f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), .8f);
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = -.875f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), .9f);
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = -1.f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), 1.f);
+
+	control_sp(ActuatorEffectiveness::ControlAxis::THRUST_Z) = -1.1f;
+	helicopter.updateSetpoint(control_sp, 0, actuator_sp);
+	EXPECT_FLOAT_EQ(actuator_sp(0), 1.f);
```

Comment:
Test is not passing since the throttle curve is defined as a constant 1, hence the returned result for actuator setpoint is always 1.

![image](https://user-images.githubusercontent.com/23277211/187684440-ac1a672b-b765-4d3a-98c9-590f383db139.png)

Reply:
Yeah, I just saw that as well. Need to configure it for a non-constant throttle curve to have a useful test.

Reply:
Fixed in b017d2b